Fix failing Docker build

FRosner opened the pull request:
https://travis-ci.org/FRosner/cluster-broccoli/jobs/262126217

```
Executing: gpg --ignore-time-conflict --no-options --no-default-keyring --homedir /tmp/tmp.qgDLXuk8pc --no-auto-check-trustdb --trust-model always --primary-keyring /etc/apt/trusted.gpg --keyring /etc/apt/trusted.gpg.d/debian-archive-jessie-automatic.gpg --keyring /etc/apt/trusted.gpg.d/debian-archive-jessie-security-automatic.gpg --keyring /etc/apt/trusted.gpg.d/debian-archive-jessie-stable.gpg --keyring /etc/apt/trusted.gpg.d/debian-archive-squeeze-automatic.gpg --keyring /etc/apt/trusted.gpg.d/debian-archive-squeeze-stable.gpg --keyring /etc/apt/trusted.gpg.d/debian-archive-wheezy-automatic.gpg --keyring /etc/apt/trusted.gpg.d/debian-archive-wheezy-stable.gpg --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys 58118E89F3A912897C070ADBF76221572C52609D
gpg: requesting key 2C52609D from hkp server p80.pool.sks-keyservers.net
gpg: keyserver timed out
gpg: keyserver receive failed: keyserver error
```

FRosner committed:
use highly available key server

https://askubuntu.com/questions/720517/key-server-times-out-while-installing-docker-on-ubuntu-14-04

swsnr commented:
I'm not sure whether that's the way forward; the [official installation instructions](https://docs.docker.com/engine/installation/linux/docker-ce/ubuntu/#install-using-the-repository) for docker look very different.  They use a different repo, and install a package called `docker-ce`.

Why are we not following the official instructions here?  Where does the current installation procedure come from originally?

FRosner committed:
install docker using the official way

https://docs.docker.com/engine/installation/linux/docker-ce/ubuntu/#install-using-the-convenience-script

swsnr reviewed (approved):
Looks much simpler :+1:

Let's see if it works.
